static/js/publisher/release/components/releasesConfirmDetails.js

```diff
@@ -0,0 +1,306 @@
+};
+
+const _ProgressiveRow = ({
+  release,
+  updateProgressiveReleasePercentage,
+  type
+}) => {
+  if (!release.progressive) {
+    return false;
+  }
+
+  let startingPercentage = 100;
+  switch (type) {
+    case progressiveTypes.RELEASE:
+      startingPercentage = release.progressive.percentage;
+      break;
+    case progressiveTypes.UPDATE:
+      startingPercentage = release.revision.release.progressive.percentage;
+      break;
+    default:
+  }
+
+  const [percentage, setPercentage] = useState(
+    release.progressive ? release.progressive.percentage : 100
+  );
+  const revisionInfo = release.revision;
+  const channel = release.channel;
+  const updatePercentage = percentage => {
+    setPercentage(percentage);
+    updateProgressiveReleasePercentage(release.progressive.key, percentage);
+  };
+
+  let progress;
+  if (
+    type === progressiveTypes.UPDATE &&
+    release.progressive.changes.some(change => change.key === "paused")
+  ) {
+    const paused = release.progressive.changes.find(
+      change => change.key === "paused"
+    ).value;
+    progress = (
+      <Fragment>
+        <ProgressiveBar percentage={percentage} />
+        <span>{paused ? "Paused" : "Resumed"}</span>
+      </Fragment>
+    );
+  } else {
+    progress = (
+      <Fragment>
+        <InteractiveProgressiveBar
+          percentage={startingPercentage}
+          onChange={updatePercentage}
+          targetPercentage={percentage}
+          minPercentage={1}
+          singleDirection={type === progressiveTypes.UPDATE ? 1 : 0}
+        />
+        <span>
+          <span className="p-tooltip--btm-right">
+            <span className="p-help">{percentage}% of devices</span>
+            <span className="p-tooltip__message">
+              Releases are delivered to devices via snap refreshes, as such, it
+              may
+              <br />
+              take some time for devices to receive the new version. There is
+              also no
+              <br />
+              guarentee that this release will achieve the entire target
+              percentage.
+            </span>
+          </span>
+        </span>
+      </Fragment>
+    );
+  }
+
+  let notes;
+  if (release.previousRevisions) {
+    const prevRev = release.previousRevisions[0].revision;
+    const prevVer = release.previousRevisions[0].version;
+
+    notes = `${100 -
+      percentage}% of devices will stay on ${prevRev} (${prevVer})`;
+  }
+
+  const displayType = type.charAt(0).toUpperCase() + type.slice(1);
+
+  return (
+    <ReleaseRow
+      type={displayType}
+      revisionInfo={revisionInfo}
+      channel={channel}
+      progress={progress}
+      notes={notes}
+    />
+  );
+};
+
+_ProgressiveRow.propTypes = {
+  release: PropTypes.object,
+  type: PropTypes.string,
+  updateProgressiveReleasePercentage: PropTypes.func
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    updateProgressiveReleasePercentage: (key, percentage) =>
```

Comment:
If this is the only prop passed via redux, maybe it can be passed via props instead to keep the component independent.

Reply:
Now I've split the component into multiple it would be a bit of a chain of props `releasesConfirm` -> `releasesConfirmDetails` -> `progressiveRow`, so I'd rather keep it in the `progressiveRow` right next to where it's used.

What do you think?

Reply:
Makes sense. Now when the component is in separate file and `connect` is done inline with export it's ok to keep it.